src/pairinteraction_gui/worker.py

```diff
@@ -88,17 +88,32 @@ def terminate_all_threads(cls) -> None:
 
 class MultiProcessWorker:
     _mp_functions_dict: ClassVar[dict[str, Callable[..., Any]]] = {}
-    all_processes: ClassVar[set["Process"]] = set()
+    _pool: ClassVar[Optional[Pool]] = None
 
     def __init__(self, fn_name: str, *args: Any, **kwargs: Any) -> None:
         if fn_name not in self._mp_functions_dict:
             raise ValueError(f"Function {fn_name} is not registered.")
 
+        if self.pool is None:
+            self.create_pool()
+
         self.fn_name = fn_name
         self.args = args
         self.kwargs = kwargs
 
-        self.queue: SimpleQueue[Any] = SimpleQueue()
+    @property
+    def pool(self) -> Pool:
+        if self._pool is None:
+            return self.create_pool()
+        return self._pool
+
+    @classmethod
+    def create_pool(cls, n_processes: int = 1) -> Pool:
```

Comment:
Why is this class method needed? Wouldn't it be sufficient to add n_processes as an argument to the constructor?